src/adapter/src/catalog.rs

```diff
@@ -973,6 +973,15 @@ impl CatalogState {
         }
     }
 
+    fn rename_cluster(&mut self, id: ClusterId, to_name: String) {
+        let cluster = self.clusters_by_id.get_mut(&id).expect("Missing cluster");
```

Comment:
We've got a purpose-built helper for this that includes the missing cluster ID in the panic message.

```suggestion
        let cluster = self.get_cluster_mut(&id);
```